### PR TITLE
[SPARK-32831][SS] Refactor SupportsStreamingUpdate to represent actual meaning of the behavior

### PR DESCRIPTION
**Pull request by HeartSaVioR**
### What changes were proposed in this pull request?

This PR renames `SupportsStreamingUpdate` to `SupportsStreamingUpdateAsAppend` as the new interface name represents the actual behavior clearer. This PR also removes the `update()` method (so the interface is more likely a marker), as the implementations of `SupportsStreamingUpdateAsAppend` should support append mode by default, hence no need to trigger some flag on it.

### Why are the changes needed?

SupportsStreamingUpdate was intended to revive the functionality of Streaming update output mode for internal data sources, but despite the name, that interface isn't really used to do actual update on sink; all sinks are implementing this interface to do append, so strictly saying, it's just to support update as append. Renaming the interface would make it clear.

### Does this PR introduce _any_ user-facing change?

No, as the class is only for internal data sources.

### How was this patch tested?

Jenkins test will follow.

### CODE REVIEW

**Comment by HeartSaVioR** [comment]
cc. @cloud-fan @rdblue @tdas

**Comment by rdblue** [comment]
Looks reasonable to me. +1

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/console.scala [hunk 1]
```diff
@@ -73,12 +73,11 @@ object ConsoleTable extends Table with SupportsWrite {
   }
 
   override def newWriteBuilder(info: LogicalWriteInfo): WriteBuilder = {
-    new WriteBuilder with SupportsTruncate with SupportsStreamingUpdate {
+    new WriteBuilder with SupportsTruncate with SupportsStreamingUpdateAsAppend {
       private val inputSchema: StructType = info.schema()
 
       // Do nothing for truncate/update. Console sink is special and it just prints all the records.
```
Comment (viirya) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/console.scala:79]: nit: remove update in the comment?

**Review by viirya — COMMENTED** [review]
Seems fine as this is only internal interface. At lease the updated doc looks clearer.

**Review by cloud-fan — APPROVED** [review]

**Comment by HeartSaVioR** [comment]
Thanks all for reviewing!

Looks like the new Jenkins build is going to be failing in KafkaOffsetReaderSuite - Jenkins build passed just before I fixed a code comment (636f050), as well as Github Actions passed. I'll consider this as build passed.

**Comment by HeartSaVioR** [comment]
Merged into master branch.

**Pull request closed without merging**